cli/root.go

```diff
@@ -59,6 +66,29 @@ func Root() *cobra.Command {
 		SilenceUsage:  true,
 		Long: `Coder — A tool for provisioning self-hosted development environments.
 `,
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if varSuppressVersion {
+				return nil
+			}
+
+			// Login handles checking the versions itself since it
+			// has a handle to an unauthenticated client.
+			if cmd.Name() == "login" {
+				return nil
+			}
+
+			client, err := createClient(cmd)
+			// If the client is unauthenticated we can ignore the check.
+			// The child commands should handle an unauthenticated client.
+			if xerrors.Is(err, errUnauthenticated) {
+				return nil
+			}
```

Comment:
We need to be authenticated to check the `/version` endpoint? Should we refactor the client to be able to start an unauthenticated client???

Might be dumb because we need the `coder url` which comes from `login` anyway in most cases. :shrug: . Just thinking out loud

Reply:
The client can start an unauthenticated client. The issue is we need the url like you said. The endpoint is not authenticated either. We just assume if we aren't logged in that it's not appropriate to compare versions (since we may not have a handle to the URL) and hand off to the child command.